pkg/ottl/ottlfuncs/func_replace_all_patterns_test.go

```diff
@@ -60,7 +60,67 @@ func Test_replaceAllPatterns(t *testing.T) {
 			function: optionalArg,
 			want: func(expectedMap pcommon.Map) {
 				expectedMap.PutStr("test", "4804d6b7f03268e33f78c484977f3d81771220df07cc6aac4ad4868102141fad world")
-				expectedMap.PutStr("test2", "4804d6b7f03268e33f78c484977f3d81771220df07cc6aac4ad4868102141fad")
+				expectedMap.PutStr("test2", "09648f12e7a3940f539bb65d32321c2f96aa94ef698d87816e94d822c6f9d7c4")
+				expectedMap.PutStr("test3", "goodbye world1 and world2")
+				expectedMap.PutInt("test4", 1234)
+				expectedMap.PutDouble("test5", 1234)
+				expectedMap.PutBool("test6", true)
+			},
+		},
+		{
+			name:    "replace only matches (with capture group and hash function)",
+			target:  target,
+			mode:    modeValue,
+			pattern: "(hello)",
```

Comment:
Can we have a test with multiple matches?

Reply:
I don't think we support hashing multiple capture groups yet. We can use them without hashing as shown in this [commit](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29408/commits/eaa490e761a5452e49de8ec979e69b9417cafee3)

Reply:
At a minimum, shouldn't we be able to use something like `$1$2` as the replacement string?

Reply:
yes we do, there's a test case for it. However we don't support hashing $1 and $2 individually. Instead we would hash the entire string "$1$2".